test/e2e/specs/widgets/editing-widgets.spec.js

```diff
@@ -0,0 +1,723 @@
+		// TODO: We can add a test for the insertion indicator here.
+		await addParagraphBlock.click();
+
+		await addedParagraphBlockInFirstWidgetArea.focus();
+		await page.keyboard.type( 'Second Paragraph' );
+
+		const addShortCodeBlock =
+			await widgetsScreen.getBlockInGlobalInserter( 'Shortcode' );
+		await addShortCodeBlock.click();
+
+		const shortCodeInput = page.getByRole( 'textbox', {
+			name: 'Shortcode text',
+		} );
+		await shortCodeInput.focus();
+		// The famous Big Buck Bunny video.
+		await shortCodeInput.type(
+			'[video src="http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4"]'
+		);
+
+		// Add to the second widget area.
+		await secondWidgetArea.click();
+		await widgetsScreen.getBlockInGlobalInserter( 'Paragraph' );
+		await addParagraphBlock.click();
+
+		const addedParagraphBlockInSecondWidgetArea =
+			secondWidgetArea.getByRole( 'document', {
+				name: /^Empty block/,
+			} );
+		await addedParagraphBlockInSecondWidgetArea.focus();
+		await page.keyboard.type( 'Third Paragraph' );
+
+		await expect.poll( widgetsScreen.getWidgetAreaBlocks ).toMatchObject( {
+			'sidebar-1': [
+				{
+					name: 'core/paragraph',
+					attributes: { content: 'First Paragraph' },
+				},
+				{
+					name: 'core/paragraph',
+					attributes: { content: 'Second Paragraph' },
+				},
+				{
+					name: 'core/shortcode',
+					attributes: {
+						text: '[video src="http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4"]',
+					},
+				},
+			],
+			'sidebar-2': [
+				{
+					name: 'core/paragraph',
+					attributes: { content: 'Third Paragraph' },
+				},
+			],
+		} );
+	} );
+
+	test( 'Should insert content using the inline inserter', async ( {
+		page,
+		widgetsScreen,
+	} ) => {
+		const firstWidgetArea = widgetsScreen.widgetAreas.first();
+
+		const inlineInserterButton = firstWidgetArea.getByRole( 'button', {
+			name: 'Add block',
+		} );
+		await inlineInserterButton.click();
+
+		const inlineQuickInserter = page.getByRole( 'listbox', {
+			name: 'Blocks',
+		} );
+
+		const paragraphBlock = inlineQuickInserter.getByRole( 'option', {
+			name: 'Paragraph',
+		} );
+		await paragraphBlock.click();
+
+		const firstParagraphBlock = firstWidgetArea.getByRole( 'document', {
+			name: /^Empty block/,
+		} );
+		await firstParagraphBlock.focus();
+		await page.keyboard.type( 'First Paragraph' );
+
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( 'Second Paragraph' );
+
+		const secondParagraphBlock = firstWidgetArea
+			.getByRole( 'document', { name: 'Block: Paragraph' } )
+			.filter( { hasText: 'Second Paragraph' } );
+
+		const secondParagraphBlockBoundingBox =
+			await secondParagraphBlock.boundingBox();
+
+		// Click outside the block to move the focus back to the widget area.
+		await firstWidgetArea.click( {
+			position: { x: 0, y: -1 },
+			// eslint-disable-next-line playwright/no-force-option
+			force: true,
```

Comment:
Nit (not a blocker): It could be helpful in future to have reason for using `force: true`

Reply:
I thought I did 😅 .

Reply:
Turns out it's not doing what it seems, I refactored it 😅 .